addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -193,12 +193,18 @@ odoo.define('website_form.s_website_form', function (require) {
                     }
                     switch (successMode) {
                         case 'redirect':
-                            successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
+                            if (successPage.startsWith(window.location.pathname + '#')) {
```

Comment:
Seeing this, you should immediately think about the langs IMHO.
I checked and indeed it won't work in translated pages, since the url is prefixed by the lang.
Maybe this should be a bit more clever, first try to see if you don't find other code occurrence that would do something like that (getting page url minus the lang).
Otherwise I guess we should do a little less robust code but that would cover the lang case, let me know if you want to discuss it.

Reply:
I fixed that in this PR. On the other hand we will still have the same kind of problem with the translated URLs. As discussed, I created a new task with this bug. [task-2754957](https://www.odoo.com/web#menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form&id=2754957)